build-tools/packages/build-cli/src/library/writeFileIfContentsDiffers.ts

```diff
@@ -0,0 +1,31 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { type PathLike } from "node:fs";
+import { readFile, writeFile } from "node:fs/promises";
+
+/**
+ * Writes to a file, but first reads the contents to check if it matches the desired content. If it does, the operation
+ * is skipped. Newlines are _always_ normalized to be line-feeds only. This pplies to both the input and the written
+ * contents. This means this function _cannot_ be used to output carriage-return/line-feed line endings.
+ *
+ * @param filePath - The path to the file to write.
+ * @param contents - The contents to write to the file.
+ * @returns True if the file was written; false otherwise.
+ */
+export async function writeFileIfContentsDiffers(
+	filePath: PathLike,
+	contents: string,
+): Promise<boolean> {
+	const fileContents = await readFile(filePath, { encoding: "utf8" });
+	const normalizedFileContents = fileContents.replace(/\r\n/g, "\n");
+	const normalizedNewContents = contents.replace(/\r\n/g, "\n");
```

Comment:
Applying this to the current file contents means that we could inadvertently leave CLRFs in the file if they somehow get there. Would it be better to only apply it to the new content? If thing differ only because of newlines, we probably want to rewrite the file anyway, I think.

Reply:
Actually, not likely that we want to rewrite. Newline difference won't matter to parsing the file for any of our uses. Git settings should be normalizing files. (There are several options for how that works.) The common thing I see with tools that don't respect the current newlines is that files is shown as changed and then when staged git fixes the file up. I see this all the time with the `feeds/public|internal*.txt` files - they show changed and then delta disappear once staged.

Reply:
@jason-ha I updated this PR to use the new function for `flub list`, too, so that should help with the feed txt files.